internal/config/config.go

```diff
@@ -19,19 +24,59 @@ type Cfg struct {
 }
 
 const (
-	cfgFile   string = "config.json"
-	cfgFolder string = "cloudstash"
+	cfgFile         string = "config.json"
+	cfgFolder       string = "cloudstash"
+	mountFolderName string = "cloudstash"
 )
 
-func DoesConfigExist(dir string) bool {
-	path := getConfigPath(dir)
+// Configure reads configuration if it already exists or creates a new one if not.
+func Configure(cfgDir, mntDir string) (cfg *Cfg, err error) {
+	if doesConfigExist(cfgDir) {
+		return parseConfig(cfgDir)
+	}
 
-	_, err := os.Stat(path)
+	cfg, err = createConfig(cfgDir, mntDir)
+	if err != nil {
+		return nil, fmt.Errorf("could not create new configuration: %v", err)
+	}
+
+	if err := writeConfig(cfgDir, cfg); err != nil {
+		return nil, fmt.Errorf("couldn't create config file: %v", err)
+	}
+
+	return cfg, nil
+}
+
+// createConfig creates a new configuration based on the given configuration & mount directories.
+func createConfig(cfgDir, mntDir string) (cfg *Cfg, err error) {
+	fmt.Print("Enter encryption secret: ")
+	pwd, err := terminal.ReadPassword(int(syscall.Stdin))
+	if err != nil {
+		return nil, fmt.Errorf("could not read encryption secret from terminal")
+	}
+
+	dbxToken, err := auth.GetDropboxToken(common.DROPBOX_APP_KEY)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't get dropbox access token: %v\n", err)
+	}
+
+	return &Cfg{
+		EncryptionKey: string(pwd),
+		MountPoint:    getMountPoint(mntDir),
+		Dropbox:       &DropboxCredentials{dbxToken},
+	}, nil
+}
 
-	return !os.IsNotExist(err)
```

Comment:
@paddlesteamer I'm getting rid of this line and assuming that the dir doesn't exist if I get **any** kind of error. It makes more sense to me this way, but feel free to revert this if I'm missing something.

Reply:
You're right, actually, this was a buggy behavior before. Now it's fixed.